### PR TITLE
Add Prometheus registerer configuration

### PR DESCRIPTION
**Pull request by povilasv**
We use a library which uses `go-metrics` and would like to have an ability to pass Prometheus registerer as an option

Example:
```
	metrics := prometheus.NewRegistry()
	metrics.MustRegister(
		version.NewCollector("thanos"),
		prometheus.NewGoCollector(),
	)

	// Memberlist uses go-metrics
	sink, err := gprom.NewPrometheusSink()
	if err != nil {
	}
	_, err = gmetrics.NewGlobal(gmetrics.DefaultConfig(cmd), sink)
	if err != nil {
            ....
	}
```

### CODE REVIEW

**Comment by povilasv** [comment]
Hey @armon maybe you could take a look at this minor change? :)

**Review by luisdavim — APPROVED** [review]
LGTM

**Comment by luisdavim** [comment]
I was about to open a similar PR...
@povilasv can you rebase?
@armon any reason not to merge this? Hot having this is actually blocking my use case...

**Comment by armon** [comment]
Merged!

File: prometheus/prometheus_test.go [hunk 1]
```diff
@@ -16,12 +17,43 @@ import (
...
+	sink, err := NewPrometheusSink()
+	if err != nil {
+		t.Fatalf("err = %v, want nil", err)
+	}
+
+	//check if register has a sink by unregistering it.
+	ok := prometheus.Unregister(sink)
+	if !ok {
+		t.Fatalf("Unregister(sink) = false, want true")
+)
```
Comment (dnephin) [inline prometheus/prometheus_test.go:55]: It looks like this is not valid Go syntax. I tried a checkout of the main branch, but this is failing to build.

I've opened a PR #114 to fix this problem.

**Pull request merged**